compute/zgbmm.c

```diff
@@ -0,0 +1,410 @@
+        plasma_error("illegal value of kl");
+        return -6;
+    }
+    if ((ku < 0) ||
+        (ku > imax(m,n)-1)) {
+        plasma_error("illegal value of ku");
+        return -7;
+    }
+
+    int am, an;
+    int bm, bn;
+    if (transa == PlasmaNoTrans) {
+        am = m;
+        an = k;
+    }
+    else {
+        am = k;
+        an = m;
+    }
+    if (transb == PlasmaNoTrans) {
+        bm = k;
+        bn = n;
+    }
+    else {
+        bm = n;
+        bn = k;
+    }
+
+    if (lda < imax(1, am)) {
+        plasma_error("illegal value of lda");
+        return -8;
+    }
+    if (ldb < imax(1, bm)) {
+        plasma_error("illegal value of ldb");
+        return -10;
+    }
+    if (ldc < imax(1, m)) {
+        plasma_error("illegal value of ldc");
+        return -13;
+    }
+
+    // quick return
+    if (m == 0 || n == 0 || ((alpha == 0.0 || k == 0) && beta == 1.0))
+        return PlasmaSuccess;
+    // Tune parameters.
+    if (plasma->tuning)
+        plasma_tune_gbmm(plasma, PlasmaComplexDouble, m, n, k, kl, ku);
+
+    // Set tiling parameters.
+    int nb = plasma->nb;
+
+    // Create tile matrices.
+    plasma_desc_t A;
+    plasma_desc_t B;
+    plasma_desc_t C;
+    int retval;
+    int tku = (ku+kl+nb-1)/nb; // number of tiles in upper band above diagonal
+                               // PLASMA conventions allow extra space for tku
+                               // for this application, could replace am with
+                               // smaller number of rows (( (ku+nb-1)/nb ))
+    int tkl = (kl+nb-1)/nb;    // number of tiles in lower band below diagonal
+    int lm = (tku+tkl+1)*nb;   // reduced number of "rows" in the matrix
+
+    retval = plasma_desc_general_band_create(PlasmaComplexDouble,PlasmaGeneral,
+                                        nb,nb,
+                                        lm, an, 0, 0, am, an, kl, ku, &A);
+    if (retval != PlasmaSuccess) {
+        plasma_error("plasma_desc_general_create() failed");
+        return retval;
+    }
+    retval = plasma_desc_general_create(PlasmaComplexDouble, nb, nb,
+                                        bm, bn, 0, 0, bm, bn, &B);
+    if (retval != PlasmaSuccess) {
+        plasma_error("plasma_desc_general_create() failed");
+        plasma_desc_destroy(&A);
+        return retval;
+    }
+    retval = plasma_desc_general_create(PlasmaComplexDouble, nb, nb,
+                                        m, n, 0, 0, m, n, &C);
+    if (retval != PlasmaSuccess) {
+        plasma_error("plasma_desc_general_create() failed");
+        plasma_desc_destroy(&A);
+        plasma_desc_destroy(&B);
+        return retval;
+    }
+
+    // Initialize sequence.
+    plasma_sequence_t sequence;
+    retval = plasma_sequence_init(&sequence);
+
+    // Initialize request.
+    plasma_request_t request;
+    retval = plasma_request_init(&request);
+
+    // asynchronous block
+    // if considering debugging , remove these directives.
```

Comment:
Please remove this comment. No information about debugging please.

Reply:
Done!